### PR TITLE
Adds titanic example to match diy-nn-with-data

### PR DESCRIPTION
**Pull request by joeyklee**
Example to be paired with diy-nn-with-data cc/ @shiffman

### CODE REVIEW

File: p5js/NeuralNetwork/NeuralNetwork_titanic/sketch.js [hunk 1]
```diff
@@ -1,45 +1,38 @@
...
-let data;
-let normalizedData;
+let nn;
 let submitButton;
+let nnOptions = {
+  dataUrl: 'data/titanic_train-experimental.csv',
+  inputs: ['age', 'fare', 'is_female'],
+  outputs: ['survived_string'],
+  task: 'classification',
+  epochs: 50,
```
Comment (shiffman) [inline p5js/NeuralNetwork/NeuralNetwork_titanic/sketch.js]: I might leave `epochs` out of the example and just use a default. . or I think this might make more sense as an argument to the `train()` function since that's when you would be deciding total epochs?

**Comment by joeyklee** [comment]
* Adding note that we need to make sure to check the input types and do oneHot encoding for string types. 
* we can more or less just copy how it is implemented for handling the outputs. This is relevant for normalization

File: p5js/NeuralNetwork/NeuralNetwork_titanic/sketch.js [hunk 2]
```diff
@@ -1,45 +1,38 @@
...
+  batchSize: 16
```
Comment (shiffman) [inline p5js/NeuralNetwork/NeuralNetwork_titanic/sketch.js]: I would also maybe leave `batchSize` out of this example and just let it use a default for simplicity. I might also suggest moving this to:

```javascript
classifier.data.setBatchSize(16);
```

**Review by shiffman — COMMENTED** [review]
I know you probably aren't finished but just taking a peek and offering some quick thoughts! I can't resist!

File: p5js/NeuralNetwork/NeuralNetwork_co2net-training/sketch.js [hunk 3]
```diff
@@ -10,253 +10,111 @@ This example uses a callback pattern to create the classifier
...
-
-function dataLoaded(err, results){
-  data = results
-  normalizedData = neuralNetwork.normalize(data);
-  console.log(normalizedData)
-
-  neuralNetwork.train(normalizedData.inputs, normalizedData.labels, finishedTraining)
-}
-
+  nn = ml5.neuralNetwork(options, modelLoaded);
```
Comment (shiffman) [inline p5js/NeuralNetwork/NeuralNetwork_co2net-training/sketch.js:33]: This is such a tiny detail and I'm not sure what is technically correct, but I might go with `modelReady()` (or `modelCreated()` since `modelLoaded()` to me implies loading a pre-trained model?

```suggestion
  nn = ml5.neuralNetwork(options, modelReady);
```

File: p5js/NeuralNetwork/NeuralNetwork_co2net-training/sketch.js [hunk 4]
```diff
@@ -10,253 +10,111 @@ This example uses a callback pattern to create the classifier
...
+function modelLoaded(){
+  console.log(nn.data);
+  nn.data.shuffle();
+  // nn.data.data = nn.data.data.map( item => {
+  //   item.xs.population_cdp = Math.log10(item.xs.population_cdp)
+  //   item.ys.scope1_ghg_emissions_tons_co2e = Math.log10(item.ys.scope1_ghg_emissions_tons_co2e) 
+  //   return item;
+  // })
+  nn.data.normalize();
+  nn.train(finishedTraining)
```
Comment (shiffman) [inline p5js/NeuralNetwork/NeuralNetwork_co2net-training/sketch.js]: I think I'm repeating something I wrote elsewhere, but thinking about. . .

```javascript
let options = {
  epochs: 50,
  batchSize: 16,
  debug: true
}
nn.train(options, finishedTraining)

File: p5js/NeuralNetwork/NeuralNetwork_titanic/sketch.js [hunk 5]
```diff
@@ -49,7 +43,7 @@ function predict() {
   let is_female = parseInt(select('#is_female').value());
 
   let inputs = [age, fare, is_female];
-  classifier.predict(inputs, function (err, results) {
+  nn.predict(inputs, function (err, results) {
```
Comment (shiffman) [inline p5js/NeuralNetwork/NeuralNetwork_titanic/sketch.js]: Not sure what to do here, but should we allow for `classify()` to be called instead of `predict()` when it's a "classification" task? Same function but multiple names? Or be more strict about it? Or just live with `predict()` only for DIY network?

**Review by shiffman — COMMENTED** [review]
can't wait to try porting my color classifier with this!

File: p5js/NeuralNetwork/NeuralNetwork_titanic/sketch.js (hunk 1 above)
Comment (joeyklee) [reply p5js/NeuralNetwork/NeuralNetwork_titanic/sketch.js]: Cool! Yep, this is resolved.

File: p5js/NeuralNetwork/NeuralNetwork_titanic/sketch.js (hunk 2 above)
Comment (joeyklee) [reply p5js/NeuralNetwork/NeuralNetwork_titanic/sketch.js]: Yep this is also resolved.

we do:

```js
const trainingOptions ={
 batchSize: 50,
epochs: 100
}
nn.train(options, finishedTraining);
```

File: p5js/NeuralNetwork/NeuralNetwork_co2net-training/sketch.js (hunk 3 above)
Comment (joeyklee) [reply p5js/NeuralNetwork/NeuralNetwork_co2net-training/sketch.js:33]: Yep this is also resolved.

File: p5js/NeuralNetwork/NeuralNetwork_titanic/sketch.js (hunk 5 above)
Comment (joeyklee) [reply p5js/NeuralNetwork/NeuralNetwork_titanic/sketch.js]: Yep this is resolved. I haven't separated out the {labels,confidence} but that is coming soon.

**Comment by joeyklee** [comment]
I will merge in these examples now and we can sync up.
